torch/csrc/utils/device.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <cstdint>
+
+namespace torch {
+
+enum class DeviceType {CPU=0, CUDA=1};
+
+struct Device {
+  const DeviceType device_type;
+  const int64_t device_index;
+  const bool is_default;   // is default device for type.
+  const int autogpu_index;
```

Comment:
Maybe we should just add an overload for `AutoGPU(const Device& dev)`? Or did you do it this way, just because it's a smaller diff right now? It would make sense to change that in the future I think

Reply:
I think I'm just going to kill this field and add 'deviceInt64' or 'deviceAutoGPUIdx' to the parser so we can avoid the r.device(0).device_index stuff.  Moving the API to AutoGPU rather than index based isn't a bad idea, but the existing code uses indices and I'd rather not change it everywhere.

Reply:
Yep, makes sense!